core/types/celo_block.go

```diff
@@ -101,11 +106,11 @@ func (h *Header) EncodeRLP(w io.Writer) error {
 			Extra:       h.Extra,
 		}
 
-		return rlp.Encode(w, &encodedHeader)
+		return encodedHeader.EncodeRLP(w)
```

Comment:
Again we can use `rlp.Encode` instead of calling `EncodeRLP` directly. 
```suggestion
		return rlp.Encode(w, &encodedHeader)
```